# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546667707 28800
#      Fri Jan 04 21:55:07 2019 -0800
# Branch stable
# Node ID 6dfb842362ddbf74c27475f0689f565fc23b7a23
# Parent  3d1a335fb9cefa2f937c2b27f97b8c84cbf1ef2e
text-renderer.h: Fix static analyzer detected issues (bug #55347).

* text-renderer.h (string (const string& s)): Initialize z data member from s.
* text-renderer.h (string operator =): Initialize z data member.

diff --git a/libinterp/corefcn/text-renderer.h b/libinterp/corefcn/text-renderer.h
--- a/libinterp/corefcn/text-renderer.h
+++ b/libinterp/corefcn/text-renderer.h
@@ -130,30 +130,31 @@ namespace octave
 
       string (const std::string& s, font& f, const double x0, const double y0)
         : str (s), family (f.get_name ()), fnt (f), x (x0), y (y0), z (0.0),
           xdata (), code (0), color (Matrix (1,3,0.0))
       { }
 
       string (const string& s)
         : str (s.str), family (s.family), fnt (s.fnt), x (s.x), y (s.y),
-          xdata (s.xdata), code (s.code), color (s.color)
+          z (s.z), xdata (s.xdata), code (s.code), color (s.color)
       { }
 
       ~string (void) = default;
 
       string& operator = (const string& s)
       {
         if (&s != this)
           {
             str = s.str;
             family = s.family;
             fnt = s.fnt;
             x = s.x;
             y = s.y;
+            z = s.z;
             xdata = s.xdata;
             code = s.code;
             color = s.color;
           }
 
         return *this;
       }
 
